pkg/apis/core/validation/validation.go

```diff
@@ -4154,6 +4152,50 @@ func validateNodeConfigSource(source *core.NodeConfigSource, fldPath *field.Path
 	return allErrs
 }
 
+// validation specific to Node.Spec.ConfigSource.ConfigMap
+func validateConfigMapNodeConfigSourceSpec(source *core.ConfigMapNodeConfigSource, fldPath *field.Path) field.ErrorList {
```

Comment:
Is there any reason separating this validation of a single object into two separate methods here since both are same spec related?

Reply:
1) You're right, it would be cleaner to remove the `Spec` suffix here, but it'll just get added back in #63314, when NodeConfigSource starts being used as part of the structured status.
2) The split between `validateNodeConfigSourceSpec` and `validateConfigMapNodeConfigSourceSpec` recognizes that the former's target (NodeConfigSource) acts as a disjoint union type, and the latter's target (ConfigMapNodeConfigSource) is one of the possible subtypes (there is only one subtype today - a ConfigMap source, but it's open to extension in the future).

Reply:
Ok. I can live with it. :-)